### PR TITLE
Standardize on using `image` in all pipelines

### LINKED ISSUE #1257: Standardize on `init_image` for img2img, inpaint, and others
**What API design would you like to have changed or added to the library? Why?**

[img2img](https://github.com/huggingface/diffusers/blob/3d98dc763a645423f992b3546c4d9cdd97eb9d04/src/diffusers/pipelines/stable_diffusion/pipeline_stable_diffusion_img2img.py#L312) currently uses the `init_image` parameter, whereas  [inpaint](https://github.com/huggingface/diffusers/blob/3d98dc763a645423f992b3546c4d9cdd97eb9d04/src/diffusers/pipelines/stable_diffusion/pipeline_stable_diffusion_inpaint.py#L312) uses the `image` parameter.

**What use case would this enable or better enable? Can you give us a code example?**

I have some code that dynamically switches between `StableDiffusionPipeline`, `StableDiffusionImg2ImgPipeline`, and `StableDiffusionInpaintPipeline` based on whether an image and mask are provided, and standardizing on `init_image` would simplify the pipeline call (instead of checking beforehand / supplying unused parameters):

```sh
pipeline(prompt, init_image=image, ...)
```

### PR DESCRIPTION
**Pull request by fboulnois**
This standardizes the use of the argument `image` in all pipelines instead of a mix of `init_image` and `image`.

Resolves #1257

### CODE REVIEW

**Comment by fboulnois** [comment]
I'm not totally sure about the code quality check errors. Running `black` and `make fix-copies` locally I see no change.

File: src/diffusers/pipelines/latent_diffusion/pipeline_latent_diffusion_superresolution.py [hunk 1]
```diff
@@ -66,7 +66,7 @@ def __init__(
     @torch.no_grad()
     def __call__(
         self,
-        init_image: Union[torch.Tensor, PIL.Image.Image],
+        image: Union[torch.Tensor, PIL.Image.Image],
```
Comment (patrickvonplaten) [inline src/diffusers/pipelines/latent_diffusion/pipeline_latent_diffusion_superresolution.py:69]: we will have to deprecate `init_image` here to not surprise the user here too much. Could you implement the same logic we're using for schedulers for all pipelines: 
https://github.com/huggingface/diffusers/blob/5755d16868ec3da7d5eb4f42db77b01fac842ea8/src/diffusers/pipelines/ddpm/pipeline_ddpm.py#L76

Here would should do the following in addition to renaming `init_image` to `image`:
1. Add a `**kwargs` argument to the function call
2. 
```python
 message = (
    "Please make sure to use `image` instead of `init_image`."
)
init_image = deprecate("init_image", "0.12.0", message, take_from=kwargs)
image = init_image or image
```

Comment (patrickvonplaten) [reply src/diffusers/pipelines/latent_diffusion/pipeline_latent_diffusion_superresolution.py:69]: We should do this for all pipelines :-)

**Review by patrickvonplaten — COMMENTED** [review]
Super nice PR @fboulnois !

Everything looks very nice to me except for deprecating `init_image` correctly. 
Could you try to implement https://github.com/huggingface/diffusers/pull/1361/files#r1033477098 for all pipelines (src/pipelines) + community? 

Let me know if you need any help :-)

**Comment by fboulnois** [comment]
@patrickvonplaten Done! Rebased on main, added deprecation warnings to community and core pipelines, and all tests pass.

File: src/diffusers/pipelines/alt_diffusion/pipeline_alt_diffusion_img2img.py [hunk 2]
```diff
@@ -540,6 +541,10 @@ def __call__(
             list of `bool`s denoting whether the corresponding generated image likely represents "not-safe-for-work"
             (nsfw) content, according to the `safety_checker`.
         """
+        message = "Please use `image` instead of `init_image`."
+        init_image = deprecate("init_image", "0.12.0", message, take_from=kwargs)
```
Comment (patrickvonplaten) [inline src/diffusers/pipelines/alt_diffusion/pipeline_alt_diffusion_img2img.py:545]: great!

**Review by patrickvonplaten — APPROVED** [review]
Super clean! Great job @fboulnois :partying_face: 

Merging :-)
